src/main/java/com/dao/homework/dao/jdbc/ManufacturerJdbcDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.dao.homework.dao.jdbc;
+
+import com.dao.homework.dao.ManufacturerDao;
+import com.dao.homework.exceptions.DataProcessingException;
+import com.dao.homework.lib.Dao;
+import com.dao.homework.model.Manufacturer;
+import com.dao.homework.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerJdbcDaoImpl implements ManufacturerDao {
+    private static final String TABLE_NAME = "manufacturer";
+    private static final String MANUFACTURER_NAME = "manufacturer_name";
+    private static final String MANUFACTURER_COUNTRY = "manufacturer_country";
+    private static final String MANUFACTURER_ID = "manufacturer_id";
+    private static final String DELETED = "deleted";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers "
+                + "(" + MANUFACTURER_NAME + "," + MANUFACTURER_COUNTRY + ") VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("This data can't be added to table ", e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        String query = "SELECT * FROM " + TABLE_NAME + " WHERE " + MANUFACTURER_ID
+                + "= ? AND " + DELETED + " = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                Manufacturer manufacturer = new Manufacturer(resultSet.getString(MANUFACTURER_ID),
+                        resultSet.getString(MANUFACTURER_NAME));
```

Comment:
A mistake from the checklist "Try to avoid code duplication. Especially, when you are working with ResultSet. Move retrieving data from ResultSet into Entity object to a separate private method."